diff --git a/src/tools/moc/moc.cpp b/src/tools/moc/moc.cpp
index 9309db1..5137654 100644
--- a/src/tools/moc/moc.cpp
+++ b/src/tools/moc/moc.cpp
@@ -531,11 +531,10 @@ void Moc::parse()
                     } else if (!test(SEMIC)) {
                         NamespaceDef def;
                         def.name = lexem();
-                        next(LBRACE);
+                        while(next()!=LBRACE);
                         def.begin = index - 1;
                         until(RBRACE);
                         def.end = index;
-                        index = def.begin + 1;
                         namespaceList += def;
                         index = rewind;
                     }
